### PR TITLE
Update benchee version

### PR DESCRIPTION
**Pull request by devonestes**
This updates this formatter to use the new behavior implementation for formatters.

### CODE REVIEW

File: .credo.exs [hunk 1]
```diff
@@ -0,0 +1,159 @@
+# This file contains the configuration for Credo and you are probably reading
+# this after creating it with `mix credo.gen.config`.
+#
+# If you find anything wrong or unclear in this file, please report an
+# issue on GitHub: https://github.com/rrrene/credo/issues
+#
```
[the remark this answers is not in the document]
Comment (PragTob) [reply .credo.exs]: I'm just guessing you copied over the main benchee config now :D

File: lib/benchee/formatters/json.ex [hunk 2]
```diff
@@ -1,12 +1,9 @@
 # Jason support for benchee structs
 require Protocol
-Enum.each([Benchee.Statistics], fn benchee_module ->
-  Protocol.derive(Jason.Encoder, benchee_module)
-end)
 
-defmodule Benchee.Formatters.JSON do
-  use Benchee.Formatter
+Protocol.derive(Jason.Encoder, Benchee.Statistics)
```
Comment (PragTob) [inline lib/benchee/formatters/json.ex]: :dancer:

File: lib/benchee/formatters/json.ex [hunk 3]
```diff
@@ -59,43 +57,44 @@ defmodule Benchee.Formatters.JSON do
...
-           |> Map.new
-    {data, filename}
+  @spec format(Suite.t(), map) :: %{Suite.key() => String.t()}
+  def format(%Suite{scenarios: scenarios}, %{file: _}) do
+    scenarios
+    |> Enum.group_by(fn scenario -> scenario.input_name end)
+    |> Enum.map(fn {input, scenarios} ->
+      {input, format_scenarios_for_input(scenarios)}
+    end)
+    |> Map.new()
```
Comment (PragTob) [inline lib/benchee/formatters/json.ex]: Love how this transforms into a full pipe as a side effect of the change :muscle:

File: test/benchee/formatters/json_test.exs [hunk 4]
```diff
@@ -1,144 +1,79 @@
 defmodule Benchee.Formatters.JSONTest do
   use ExUnit.Case
-  import ExUnit.CaptureIO
   doctest Benchee.Formatters.JSON
 
-  test ".output returns the suite again unchanged" do
```
Comment (PragTob) [inline test/benchee/formatters/json_test.exs]: we seem to have lost this test?

File: .credo.exs (hunk 1 above)
[the remark this answers is not in the document]
Comment (devonestes) [reply .credo.exs]: Yeah, for most stuff I just copied over from Benchee.

File: test/benchee/formatters/json_test.exs (hunk 4 above)
Comment (devonestes) [reply test/benchee/formatters/json_test.exs]: We no longer have `output/1` as a callback for formatters - just `format/2` and `write/2`, so I deleted the function and the associated test.

Comment (PragTob) [reply test/benchee/formatters/json_test.exs]: ah right ase output is now assured in the formatter itself. Am/was tired and got ya, thanks!

**Pull request merged**